app/models/recently_added.rb

```diff
@@ -24,5 +19,21 @@ def self.feed(root_path)
       }
     end
     payload
+  rescue StandardError => ex
+    Rails.logger.warn "Error fetching recently added feed: #{ex.message}."
+    {}
+  end
+
+  def self.http_get(url, read_timeout: 3)
+    uri = URI.parse(url)
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.read_timeout = read_timeout
```

Comment:
This is what I added to force a short timeout value (we are defaulting it to 3 seconds on line 27)